tools/src/h5dump/h5dump.c

```diff
@@ -1471,6 +1471,10 @@ main(int argc, const char *argv[])
         if (table_list_add(fid, oi.fileno) < 0) {
             error_msg("internal error (file %s:line %d)\n", __FILE__, __LINE__);
             h5tools_setstatus(EXIT_FAILURE);
+            /* table_list.nused will be zero and the added containers needs to be cleaned up */
+            HDfree(table_list.tables[0].group_table);
+            HDfree(table_list.tables[0].dset_table);
+            HDfree(table_list.tables[0].type_table);
             goto done;
```

Comment:
I'm not very familiar with the code, but, generally, I think freeing should be done in done:, also, should table_list_free() be used somehow?

Reply:
Normally, yes, but these are pointers not captured anywhere else until later.

Reply:
I assume that since the sanitizer reported an issue, table_list_free() in done wasn't freeing these group, dset, and type tables belonging to table_list.tables[0]?  Should table_list_free() in done (line 1614) have freed them?  Is there a need to check that they exist before freeing them here?

Reply:
I agree.  If these lines stay there, the pointers need to be checked before freeing.